packages/sst/src/constructs/RDS.ts

```diff
@@ -393,7 +403,7 @@ export class RDS extends Construct implements SSTConstruct {
     }
 
     // Validate Secrets Manager is used for "credentials"
-    if (props.credentials && !props.credentials.secret) {
+    if (props.credentials && !props.credentials.secret && props.credentials.password) {
```

Comment:
Why is this required? Can a credential have both `secret` and `password`?

Reply:
The original check forbad using `credentials` in `cdk`. The example in the test was providing credentials with your own password. This would not create a secret. If you specify credentials differently, a secret is created, but later, so at this point it is not set. But password is also not set. So I added this check to still pass the initial test, but allow the latter case. Hopefully this is clear? The way I provide credentials creates that secret.

This is what it looks like in CF:

```yaml
  RDSE0E96D00:
    Type: AWS::RDS::DBCluster
    Properties:
      CopyTagsToSnapshot: true
      DBClusterParameterGroupName: default.aurora-postgresql11
      Engine: aurora-postgresql
      EngineMode: serverless
      EngineVersion: "11.16"
      MasterUsername:
        Fn::Join:
          - ""
          - - "{{resolve:secretsmanager:"
            - Ref: RDSSecret3683CA93
            - :SecretString:username::}}
      MasterUserPassword:
        Fn::Join:
          - ""
          - - "{{resolve:secretsmanager:"
            - Ref: RDSSecret3683CA93
            - :SecretString:password::}}
      StorageEncrypted: true
      VpcSecurityGroupIds: []
    UpdateReplacePolicy: Snapshot
    DeletionPolicy: Snapshot
```

The password in the example had a token with `{{resolve:ssm}}`.

Reply:
Got it! Thanks for the explanation.

Changed to this as the `props.credentials.secret` check seems to be not useful.
```diff
- if (props.credentials && !props.credentials.secret && props.credentials.password) {
+ if (props.credentials?.password) {
```